runners/spark/src/main/java/org/apache/beam/runners/spark/coders/ObjectInputStreamWithClassLoader.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.runners.spark.coders;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectStreamClass;
+
+
+/**
+ * ObjectInputStream with specific ClassLoader.
+ */
+class ObjectInputStreamWithClassLoader extends ObjectInputStream {
+  private final ClassLoader classLoader;
+
+  ObjectInputStreamWithClassLoader(InputStream in, ClassLoader classLoader) throws IOException {
+    super(in);
+    this.classLoader = classLoader;
+  }
+
+  @Override
+  protected Class<?> resolveClass(ObjectStreamClass desc) {
+    try {
+      return Class.forName(desc.getName(), false, classLoader);
+    } catch (ClassNotFoundException e) {
+      throw new RuntimeException("Could not find class: " + desc.getName(), e);
```

Comment:
`primClasses` is a private member of `ObjectInputStream`. In any case, primitives aren't serialized with `StatelessJavaSerializer` only the classes that were registered to use it (`Source` and `Coder` implementations)